devel/create_distro_tarball.sh

```diff
@@ -0,0 +1,226 @@
+#!/bin/sh
+#
+# Copyright 2011 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Author: morlovich@google.com (Maksim Orlovich)
+#
+# The usual mechanism used to develop mod_pagespeed and build binaries is based
+# on merging all dependencies into a single source tree.  This script enables a
+# standard  untar/configure/make flow that does not bundle widely available
+# external libraries. It generates the tarball including the configure (or
+# rather generate.sh) script.
+#
+# If --minimal is passed, it will cut out even more things. This was meant
+# for packaging properly Debian, which has a particularly extensive package
+# repository. At the moment this configuration requires further patching of
+# the .gyp[i] files and doesn't work out of the box. The pruning was also done
+# as of branch 33, so further tweaks might be required for this mode in
+# 34 or newer.
+
+set -e  # exit script if any command returns an error
+set -u  # exit the script if any variable is uninitialized
+
+function usage {
+  echo -n "create_distro_tarball_debian.sh [ --minimal ] [ --checkout_dir dir ] tarball"
+  echo " [ --git_tag tag | source_tree ]"
+  echo "examples of git_tag would be 'master', '33', '1.11.33.4' or 04e3237"
+  exit 1
+}
+
+# This outputs a little wrapper around gyp that calls it with appropriate -D
+# flag
+function config {
+  cat <<SCRIPT_END
+#!/bin/sh
+#
+# This script uses gyp to generate Makefiles for mod_pagespeed built against
+# the following system libraries:
+#   apr, aprutil, apache httpd headers, icu, libjpeg_turbo, libpng, zlib.
+#
+# Besides the -D use_system_libs=1 below, you may need to set (via -D var=value)
+# paths for some of these libraries via these variables:
+#   system_include_path_httpd, system_include_path_apr,
+#   system_include_path_aprutil.
+#
+# for example, you might run
+# ./generate.sh -Dsystem_include_path_apr=/usr/include/apr-1 \\
+#               -Dsystem_include_path_httpd=/usr/include/httpd
+# to specify APR and Apache include directories.
+#
+# Also, BUILDTYPE=Release can be passed to make (the default is Debug).
+echo "Generating src/Makefile"
+src/build/gyp_chromium -D use_system_libs=1 \$*
+SCRIPT_END
+}
+
+if [ $# -lt 2 ]; then
+  usage
+  exit
+fi
+
+CHECKOUT_DIR=
```

Comment:
This would be SO MUCH more readable if it used getopt. Take a look here:

https://github.com/pagespeed/mod_pagespeed/blob/d55c2d7a2c64fab183f77571a18dad80ffe3cb1b/install/build_on_vm.sh#L34